src/components/molecules/Visualizer/hooks.ts

```diff
@@ -373,6 +377,22 @@ function useProviderProps(
     [engineRef],
   );
 
+  const getLayersInViewport = useCallback(() => {
+    const rect = engineRef.current?.getViewport();
+    return layers.findAll(
+      layer =>
+        rect &&
+        layer.property &&
+        Rectangle.contains(
+          rect,
```

Comment:
```suggestion
          new Rectangle(rect.west, rect.south, rect.east, rect.north),
```

Cesium is also a specific map engine, but at the same time it is a general purpose and useful GIS library, so it is fine to use Cesium for calculations.